corehq/apps/hqcase/tests/test_case_api_permissions.py

```diff
@@ -0,0 +1,213 @@
+import uuid
+from contextlib import contextmanager
+
+from django.test import TestCase
+from django.urls import reverse
+
+from casexml.apps.case.mock import CaseBlock
+
+from corehq import privileges
+from corehq.apps.commtrack.tests.util import bootstrap_domain
+from corehq.apps.es.tests.utils import (
+    case_search_es_setup,
+    case_search_es_teardown,
+    es_test,
+)
+from corehq.apps.hqcase.api.get_list import get_list
+from corehq.apps.locations.tests.util import setup_locations_and_types
+from corehq.apps.users.models import HqPermissions, UserRole, WebUser
+from corehq.elastic import get_es_new
+from corehq.pillows.mappings.user_mapping import USER_INDEX
```

Comment:
These are the "old" way. More comments below. If you're curious to look further, @AmitPhulera recently modernized existing tests, and his PR (#32522) is a great reference for becoming familiar with the new technique.

```suggestion
```